src/test/java/com/commercetools/sync/producttypes/ProductTypeSyncTest.java

```diff
@@ -447,6 +450,79 @@ void sync_WithInvalidAttributeDefinitions_ShouldThrowError() {
         assertThat(productTypeSyncStatistics).hasValues(1, 0, 0, 1, 0);
     }
 
+    @Test
+    void sync_WithErrorOnResolvingMissingReference_ShouldThrowError() {
+        String draftKey = "key2";
+        // preparation
+        final ProductTypeDraft newProductTypeDraft2 = ProductTypeDraft.ofAttributeDefinitionDrafts(
+            draftKey,
+            "name",
+            "desc",
+            emptyList()
+        );
+        NestedAttributeType nestedTypeAttrDefDraft1 = NestedAttributeType
+            .of(ProductType.referenceOfId(draftKey));
+        final AttributeDefinitionDraft nestedTypeAttrDefDraft = AttributeDefinitionDraftBuilder
+            .of(nestedTypeAttrDefDraft1, "validNested", ofEnglish("koko"), true)
+            .build();
+
+        // preparation
```

Comment:
You have twice preparation but no `// test` :)

Reply:
c1816e70